### PR TITLE
png2src: order palette

### PR DESCRIPTION
**Pull request by JerwuQu**
For #358

Please give it a go and see if there are any strange surprises

### CODE REVIEW

**Review by christopher-kleine — APPROVED** [review]
I checked the source of this PR. I also checked the result.
Here is the result:
| Case                 | Old     | Current | This PR |
|----------------------|---------|---------|---------|
| Indexed (5/5 Colors)   | Invalid | Invalid | Invalid |
| RGB (5 Colors)       | Invalid | Invalid | Invalid |
| Indexed (2/2 Colors) | 1 Bit   | 1 Bit   | 1 Bit   |
| Indexed (2/4 Colors) | 2 Bit   | 1 Bit   | 1 Bit |
| Indexed (2/5 Colors) | Invalid | 1 Bit   | 1 Bit |
| RGB (2 Colors)       | Invalid | 1 Bit   | 1 Bit   |
| Indexed (4/4 Colors) | 2 Bit   | 2 Bit   | 2 Bit |
| Indexed (4/5 Colors) | Invalid | 2 Bit   | 2 Bit |
| RGB (4 Colors)       | Invalid | 2 Bit   | 2 Bit   |

Special Case: Swapping index in index images. This PR handles this correctly.
And while I'd like to have a flag to ignore the index if I want, it's not needed in like 99% of the cases.

My opinion: Well done. You've beat me to the punch here. :smile:

File: cli/lib/png2src.js [hunk 1]
```diff
@@ -130,31 +130,50 @@ function run(sourceFile) {
...
-                    g: g,
-                    b: b,
-                    a: a,
+                palette.set(packed, {
+                    r, g, b, a, x, y,
                     i: colorCount++,
-                    x: x,
-                    y: y,
-                };
+                    brightness: (r + g + b) * a,
```
Comment (aduros) [inline cli/lib/png2src.js:145]: Would it be better to use a perceptual brightness here? eg. luma:

```suggestion
                    brightness: (0.2126*r + 0.7152*g + 0.0722*b) * a,
```

Comment (JerwuQu) [reply cli/lib/png2src.js]: Sure thing! The "Commit suggestion" button on Github was neat, thank you!

**Comment by aduros** [comment]
Looks perfect, thanks  a lot!

**Pull request merged**